File: LogicalThinkingPatterns/BinaryNumberTrianglePattern/BinaryNumberTriangle.py
class BinaryNumberTriangle:
    def printPattern(self, n):
        for i in range(0, n):
            pattern = ""
            for j in range(0,i+1):
                if (i % 2 == 0):
                    if (j % 2 == 0):
                        pattern += "1 "
                    else:
                        pattern += "0 "
                else:
                    if(j % 2 == 0):
                        pattern += "0 "
                    else:
                        pattern += "1 "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

binaryNumberTriangle = BinaryNumberTriangle()
binaryNumberTriangle.run()

File: LogicalThinkingPatterns/BinaryNumberTrianglePattern/BinaryNumberTriangle.ts
class BinaryNumberTriangle {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = 0; j <= i; j++) {
        if (i % 2 == 0) {
          if (j % 2 == 0) {
            pattern += "1 ";
          } else {
            pattern += "0 ";
          }
        } else {
          if (j % 2 == 0) {
            pattern += "0 ";
          } else {
            pattern += "1 ";
          }
        }
      }
      console.log(pattern);
    }
  }
}

const binaryNumberTriangle = new BinaryNumberTriangle();
binaryNumberTriangle.printPattern(5);
